x-pack/platform/plugins/shared/fleet/server/services/agents/status.ts

```diff
@@ -195,99 +191,70 @@ export async function getIncomingDataByAgentsId({
   const logger = appContextService.getLogger();
 
   try {
-    const dataStreamPatternChunks = chunk(
-      dataStreamPattern.split(','),
-      DATASTREAM_PATTERN_CHUNK_SIZE
-    );
-
-    for (const datastreamChunck of dataStreamPatternChunks) {
-      const { has_all_requested: hasAllPrivileges } = await esClient.security.hasPrivileges({
-        index: [
-          {
-            names: [datastreamChunck.join(',')],
-            privileges: ['read'],
-          },
-        ],
-      });
+    const { has_all_requested: hasAllPrivileges } = await esClient.security.hasPrivileges({
+      index: [
+        {
+          names: dataStreamPattern.split(','),
```

Comment:
is there any limit to the length of names that we should consider?

Reply:
I tested with a larger number 200 datastreams and it seems there is no issue, currently our largest package aws has 46 datastreams so it seems safe to me.